src/amo/pages/Addon/index.js

```diff
@@ -532,6 +532,7 @@ export class AddonBase extends React.Component {
                 {showInstallButton && (
                   <AMInstallButton
                     addon={addon}
+                    currentVersion={currentVersion}
```

Comment:
That is a good question. Here's what I was thinking: If I use `getVersionById` in `mapStateToProps`, it's possible I'll get `null` back, because maybe the version hasn't been loaded yet. I would therefore need logic to deal with the case where `currentVersion` is `null`. It doesn't really make sense to call this component without a `currentVersion` because if you don't have a `currentVersion` you cannot have a (working) install button, so it seemed simpler to just require a `currentVersion` as a property for this component.

Maybe I'm thinking about this the wrong way, and maybe it would be better to get the `currentVersion` from redux and also deal with the `null` logic in this component. What do you think?

Reply:
I do agree with you.